modin/core/storage_formats/pandas/parsers.py

```diff
@@ -653,9 +674,49 @@ def parse(fname, **kwargs):
         columns = [c for c in columns if c not in df.index.names and c in df.columns]
         if columns is not None:
             df = df[columns]
-        # Append the length of the index here to build it externally
         return _split_result_for_readers(0, num_splits, df) + [idx, df.dtypes]
 
+    @staticmethod
+    @doc(
+        _doc_parse_func,
+        parameters="""files_for_parser : list
+    List of files to be read.
+storage_options : dict
+    Parameters for specific storage engine.""",
+    )
+    def parse_fsspec_files(files_for_parser, storage_options, **kwargs):
+        from pyarrow.parquet import ParquetFile
+
+        columns = kwargs.get("columns", None)
+
+        chunks = []
+        for file_for_parser in files_for_parser:
+            if isinstance(file_for_parser.path, IOBase):
+                context = nullcontext(file_for_parser.path)
+            else:
+                context = fsspec.open(file_for_parser.path, **storage_options)
+            with context as f:
+                chunk = (
+                    ParquetFile(f)
+                    .read_row_groups(
+                        list(
+                            range(
+                                file_for_parser.row_group_start,
+                                file_for_parser.row_group_end,
+                            )
+                        ),
+                        columns=columns,
+                        use_pandas_metadata=True,
+                    )
+                    .to_pandas()
```

Comment:
why?

Reply:
See above - does it make sense as to what we are doing here? Did you have a different approach in mind?

Reply:
To be clear - the `to_pandas` here is converting from PyArrow to pandas (which we want the partitions to be)